REPORT.md

```diff
@@ -0,0 +1,145 @@
+
+## Complexity
+
+##### 1. What are your results for the ten most complex functions? (If ranking is not easily possible: ten complex functions)?
+The following table shows the ten most complex functions that were found by `lizard` and the computed cyclomatic complexity by lizard (CCN1) as well as the cyclomatic complexity counted by hand (CCN2). The formula used for calculating CCN2 was [M = E − N + 2P](https://en.wikipedia.org/wiki/Cyclomatic_complexity). 
+
+| ID 	| File                                                                           	| Function              	| CCN1 	| CCN2 	|
+|----	|--------------------------------------------------------------------------------	|-----------------------	|------	|------	|
+| 1 	| litiengine/src/de/gurkenlabs/litiengine/configuration/ConfigurationGroup.java  	| storeProperties       	| 19   	| 18   	|
+| 2  	| litiengine/src/de/gurkenlabs/litiengine/environment/Environment.java           	| add                   	| 18   	| 17   	|
+| 3  	| litiengine/src/de/gurkenlabs/litiengine/environment/Environment.java           	| remove                	| 18   	|      	|
+| 4  	| litiengine/src/de/gurkenlabs/litiengine/environment/MapObjectSerializer.java   	| getPropertyValue      	| 18   	| 18   	|
+| 5  	| litiengine/src/de/gurkenlabs/litiengine/environment/tilemap/MapUtilities.java  	| getTile               	| 18   	| 17   	|
+| 6  	| litiengine/src/de/gurkenlabs/litiengine/environment/CustomMapObjectLoader.java 	| invoke                	| 17   	|      	|
+| 7  	| litiengine/src/de/gurkenlabs/litiengine/graphics/RenderComponent.java          	| render                	| 17   	|      	|
+| 8  	| litiengine/src/de/gurkenlabs/litiengine/environment/tilemap/xml/Layer.java     	| afterUnmarshal        	| 16   	| 15   	|
+| 9  	| litiengine/src/de/gurkenlabs/litiengine/environment/tilemap/MapUtilities.java  	| assessHexStaggering   	| 16   	| 15   	|
+| 10 	| litiengine/src/de/gurkenlabs/litiengine/util/geom/GeometricUtilities.java      	| getIntersectionPoints 	| 16   	| 16   	|
+
+###### a. Did all tools/methods get the same result?
+The calculations we performed resulted in very similar results. However, something that the calculations had in common, compared to the results from running lizard, was that they were slightly lower in general (see function #2, #5, #8 and #9). 
+
+###### b. Are the results clear?
+The functions were relative straightforward which made it easy to graph. As such, it was somewhat clear how lizard calculated the CNN, given that the hand calculated CCN gave a similar result to lizard's CCN.   
+
+##### 2. Are the functions just complex, or also long?
+The length of the functions is varying, from around 10 lines to 75. In other words, not all complex functions in this project are very long.
```

Comment:
I think that the function that was 10 loc was the one that I accidentally thought was mine, the one that was actually mine was closer to 35.